src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -651,6 +651,8 @@ private function writeProperty($zval, $property, $value)
             // fatal error.
 
             $object->$property = $value;
+        } elseif (self::ACCESS_TYPE_NOT_FOUND === $access[self::ACCESS_TYPE]) {
+            throw new NoSuchPropertyException(sprintf('Could not determine access type for property "%s".', $property));
         } elseif (self::ACCESS_TYPE_MAGIC === $access[self::ACCESS_TYPE]) {
```

Comment:
An `elseif` expression after an uncaught exception looks weird. IMHO, it should be more semantic to replace it with a new `if`.